frontend/src/metabase-lib/queries/structured/Filter.ts

```diff
@@ -85,6 +90,55 @@ export default class Filter extends MBQLClause {
     return [undefined, undefined];
   }
 
+  /**
+   * Tries to return a DatePicker-compatible version of this filter because:
+   *   1. DatePicker cannot currently handle week/month/quarter/year units, so we convert them to day units instead.
+   *   2. DatePicker needs dates and times to be specifically formatted to detect if time component should be shown.
+   *   3. SpecificDatePicker will not display dates or times unless they match DATE_FORMAT or DATE_TIME_FORMAT.
+   */
+  toDatePickerFilter() {
```

Comment:
"DatePicker filter" is a UI concern so it would not be great to add it to the MBQL manipulation library module. This function is used only once, can we exctract it from `Filter.ts` and use in the component?